.github/workflows/ci.yml

```diff
@@ -41,14 +41,36 @@ jobs:
       - name: Setup Build Env
         run: sudo ./scripts/install-build-tools.sh
       - name: Setup Local Dependencies
-        run: ./scripts/setup-dependencies.sh
+        run: sudo ./scripts/setup-dependencies.sh
       - name: Build
-        run: scripts/build.sh
+        run: ./scripts/build.sh
       - name: Lint
-        run: scripts/lint.sh
+        run: ./scripts/lint.sh
+  pylint:
+    name: Pylint
+    runs-on: ubuntu-22.04
+    continue-on-error: true
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
```

Comment:
It does run of each of those, but I see it as valuable to have compatibility for several versions. We discussed using version 3.10 and more recent

Reply:
So are the linting standards different for these versions of Python? 

My concern if yes is we could have a proposed feature that is known to only work on Python 3.12 for example, and then it may fail on 3.10 and 3.11 even though it is known to be incompatible with those versions. Further if two of these versions have inconsistent lint checks, one could get caught in a situation where one will always pass when the other fails.

If no, then we are doing extra work and should just take the highest standard (likely 3.12)

Reply:
@HalosGhost Would value your input here as well

Reply:
@HalosGhost in the beginning I thought you were in favor of multiple versions (3.10+) so the user wouldn't have to install another python version. That's why I we had the loops in `install-build-tools.sh` to check if the user had a compatible version already.

If we are constrained to only use one version - I'd give my strong opinion for 3.12.
1) `asyncio` module gets a big performance boost in 3.12. Parallelization in python will be important as we add the py runner and more
2) 3.10 is less future-proofed with security support (2 years left instead of 4).

Lesser reasons - Improved errors and debugging. It does better on pylint. The module `distutils` is deprecated for python3.12. If we had 3.10 as the supported version then we'd have to add `setuptools` back to the requirements.txt file for installation to work (assuming we remove is as not reqd for 3.10 install)

<img width="751" alt="image" src="https://github.com/mit-dci/opencbdc-tx/assets/18053614/66b457cf-5845-4bbd-b1f6-7b3c1fcd0c0e">
https://endoflife.date/python

Reply:
After discussing with @maurermi, we've decided to have just python3.10 in the ci.yml task for pylint. If the user has any of the versions 3.10-3.12 installed, then python and pylint will also be compatible with those. @HalosGhost it looks like your comment about testing the lowest supported version supports this. Does this approach make sense? I updated the code in the most recent amended commit to loop through py versions in order of 3.10 to 3.12 instead of reverse as well. Thanks